server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java

```diff
@@ -158,6 +158,9 @@ public synchronized long getQueuedJobs() {
   }
 
   public synchronized long getLowestPriority() {
+    if (jobQueue.isEmpty()) {
```

Comment:
If this method returned an OptionalLong, could the metrics code not emit a metric when its empty?

Reply:
Each queue is registered with the MeterRegistry which calls this method directly on update. 
So the Micrometer code would have to support handling an Optional. 

Looking at their docs, I don't see anything popping out about Optionals.
We could certainly write some test cases and see what happens.

Reply:
> Looking at their docs, I don't see anything popping out about Optionals.

ok, that makes sense.  Then if this method returned an optional then when its empty we would have to unregister the metrics, which would not be pretty.